consensus/abci.go

```diff
@@ -34,8 +36,8 @@ const maxTransactionSize = 32768
 // Flow:
 //		1. BeginBlock
 //		2. CheckTx
-//	    3. DeliverTx
-//		4. EndBlock
+//		3. EndBlock
+//		4. DeliverTx*
```

Comment:
[comment] After debugging the code I noticed we got the wrong order for this

Reply:
After debugging step by step I got the that is actually the correct flow. Did you validate it  is not?